src/streamlit_extras/__init__.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+from importlib import import_module
+from typing import Any, Callable, Optional, TypeVar, Union, overload
+
+from streamlit import _gather_metrics
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+@overload
+def extra(
+    func: F,
+) -> F:
+    ...
+
+
+@overload
+def extra(
+    func: None = None,
+) -> Callable[[F], F]:
```

Comment:
I'm not quite getting what's happening here with these overloads -- why would a None ever get passed?

Reply:
Yea good point. Actually this is @LukasMasuch's part, and I think he made these corner cases just in case streamlit-extras in running with an old version of streamlit, where page profiling would not work... not sure.

Reply:
The typing overloads here are actually required so that you can correctly (= with correct typing) use the decorator in different ways: 1) as a decorator without parameters `@extra` 2) as a decorator with parameters (`@extra(foo="bar")`  but this also refers to empty parameters `@extra()` 3) as a function: `extra(my_function)`

Reply:
@LukasMasuch Thanks! Yeah, that makes sense -- I guess that's just part of the weirdness of the different ways decorators can be used.

Reply:
(Adding this to the docstring)

Reply:
Added 8392df5